packages/api/cms-api/src/dam/files/dam-upload-file.interceptor.ts

```diff
@@ -52,8 +52,19 @@ export function DamUploadFileInterceptor(fieldName: string): Type<NestIntercepto
             this.fileInterceptor = new (NestFileInterceptor(fieldName, multerOptions))();
         }
 
-        intercept(...args: Parameters<NestInterceptor["intercept"]>): Observable<unknown> | Promise<Observable<unknown>> {
-            return this.fileInterceptor.intercept(...args);
+        async intercept(context: ExecutionContext, next: CallHandler<unknown>): Promise<Observable<unknown>> {
+            const fileInterceptor = await this.fileInterceptor.intercept(context, next);
+
+            const ctx = context.switchToHttp();
+            const file = ctx.getRequest().file;
+
+            const error = await this.fileValidationService.validateFileContents(file);
+
+            if (error) {
+                return throwError(() => new HttpException("Rejected File Upload: SVG must not contain JavaScript", 422));
```

Comment:
This error message is too specific. There may be other content violations in the future. The error should be defined in validateFileContents

Reply:
True, I missed that